src/main/java/com/instaclick/pentaho/plugin/amqp/AMQPPlugin.java

```diff
@@ -515,10 +686,37 @@ public void dispose(StepMetaInterface smi, StepDataInterface sdi)
         meta = (AMQPPluginMeta) smi;
         data = (AMQPPluginData) sdi;
 
+        logDebug("DISPOSE INVOKED");
+
+        waitForConfirtmationStepsFinished();

```

Comment:
@FabioBatSilva 
removed wait, but now 
2015/06/03 18:55:57 - CONSUMER.0 - Finished processing (I=14, O=0, R=0, W=14, U=0, E=0)
diffs from 
2015/06/03 18:56:09 - CONSUMER.0 - QUEUE MESSAGES RECEIVED : ACK=4, REJECTED=6, REQUEUE=4

maybe it is OK, and should be noted in readme